<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo"  />
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Peer Assignment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a Db2 database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. Click on the links below to download and save the datasets (.CSV files):

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01" target="_blank">Chicago Census Data</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01" target="_blank">Chicago Public Schools</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01" target="_blank">Chicago Crime Data</a>

**NOTE**: For the learners who are encountering issues with loading from .csv in DB2 on Firefox, you can download the .txt files and load the data with those:

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.txt?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01" target="_blank">Chicago Census Data</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.txt?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01" target="_blank">Chicago Public Schools</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.txt?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01" target="_blank">Chicago Crime Data</a>

**NOTE:** Ensure you have downloaded the datasets using the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


### Store the datasets in database tables

To analyze the data using SQL, it first needs to be stored in the database.

While it is easier to read the dataset into a Pandas dataframe and then PERSIST it into the database as we saw in Week 3 Lab 3, it results in mapping to default datatypes which may not be optimal for SQL querying. For example a long textual field may map to a CLOB instead of a VARCHAR.

Therefore, **it is highly recommended to manually load the table using the database console LOAD tool, as indicated in Week 2 Lab 1 Part II**. The only difference with that lab is that in Step 5 of the instructions you will need to click on create "(+) New Table" and specify the name of the table you want to create and then click "Next".

<img src = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/LoadingData.png">

##### Now open the Db2 console, open the LOAD tool, Select / Drag the .CSV file for the first dataset, Next create a New Table, and then follow the steps on-screen instructions to load the data. Name the new tables as follows:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**


### Connect to the database

Let us first load the SQL extension and establish a connection with the database

The following required modules are pre-installed in the Skills Network Labs environment. However if you run this notebook commands in a different Jupyter environment (e.g. Watson Studio or Ananconda) you may need to install these libraries by removing the `#` sign before `!pip` in the code cell below.


In [1]:
# These libraries are pre-installed in SN Labs. If running in another environment please uncomment lines below to install them:
# !pip install --force-reinstall ibm_db==3.1.0 ibm_db_sa==0.3.3
# Ensure we don't load_ext with sqlalchemy>=1.4 (incompadible)
# !pip uninstall sqlalchemy==1.4 -y && pip install sqlalchemy==1.3.24
# !pip install ipython-sql

In [2]:
%load_ext sql

In the next cell enter your db2 connection string. Recall you created Service Credentials for your Db2 instance in first lab in Week 3. From your Db2 service credentials copy everything after db2:// (except the double quote at the end) and paste it in the cell below after ibm_db_sa://

<img src ="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/details.png">


In [3]:
# Remember the connection string is of the format:
# %sql ibm_db_sa://my-username:my-password@my-hostname:my-port/my-db-name?security=SSL
# Enter the connection string for your Db2 on Cloud database instance below
%sql ibm_db_sa://gxn80469:InjWDXk3dTEYyXIh@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB?security=SSL

'Connected: gxn80469@BLUDB'

In [4]:
%sql select * from "CENSUS_DATA" Limit 20;

 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[(1, 'Rogers Park', Decimal('7.7'), Decimal('23.6'), Decimal('8.7'), Decimal('18.2'), Decimal('27.5'), 23939, 39),
 (2, 'West Ridge', Decimal('7.8'), Decimal('17.2'), Decimal('8.8'), Decimal('20.8'), Decimal('38.5'), 23040, 46),
 (3, 'Uptown', Decimal('3.8'), Decimal('24.0'), Decimal('8.9'), Decimal('11.8'), Decimal('22.2'), 35787, 20),
 (4, 'Lincoln Square', Decimal('3.4'), Decimal('10.9'), Decimal('8.2'), Decimal('13.4'), Decimal('25.5'), 37524, 17),
 (5, 'North Center', Decimal('0.3'), Decimal('7.5'), Decimal('5.2'), Decimal('4.5'), Decimal('26.2'), 57123, 6),
 (6, 'Lake View', Decimal('1.1'), Decimal('11.4'), Decimal('4.7'), Decimal('2.6'), Decimal('17.0'), 60058, 5),
 (7, 'Lincoln Park', Decimal('0.8'), Decimal('12.3'), Decimal('5.1'), Decimal('3.6'), Decimal('21.5'), 71551, 2),
 (8, 'Near North Side', Decimal('1.9'), Decimal('12.9'), Decimal('7.0'), Decimal('2.5'), Decimal('22.6'), 88669, 1),
 (9, 'Edison Park', Decimal('1.1'), Decimal('3.3'), Decimal('6.5'), Decimal('7.4'), Decimal('35.3'), 40959, 8),
 (10, 'Norwood Park', Decimal('2.0'), Decimal('5.4'), Decimal('9.0'), Decimal('11.5'), Decimal('39.5'), 32875, 21),
 (11, 'Jefferson Park', Decimal('2.7'), Decimal('8.6'), Decimal('12.4'), Decimal('13.4'), Decimal('35.5'), 27751, 25),
 (12, 'Forest Glen', Decimal('1.1'), Decimal('7.5'), Decimal('6.8'), Decimal('4.9'), Decimal('40.5'), 44164, 11),
 (13, 'North Park', Decimal('3.9'), Decimal('13.2'), Decimal('9.9'), Decimal('14.4'), Decimal('39.0'), 26576, 33),
 (14, 'Albany Park', Decimal('11.3'), Decimal('19.2'), Decimal('10.0'), Decimal('32.9'), Decimal('32.0'), 21323, 53),
 (15, 'Portage Park', Decimal('4.1'), Decimal('11.6'), Decimal('12.6'), Decimal('19.3'), Decimal('34.0'), 24336, 35),
 (16, 'Irving Park', Decimal('6.3'), Decimal('13.1'), Decimal('10.0'), Decimal('22.4'), Decimal('31.6'), 27249, 34),
 (17, 'Dunning', Decimal('5.2'), Decimal('10.6'), Decimal('10.0'), Decimal('16.2'), Decimal('33.6'), 26282, 28),
 (18, 'Montclaire', Decimal('8.1'), Decimal('15.3'), Decimal('13.8'), Decimal('23.5'), Decimal('38.6'), 22014, 50),
 (19, 'Belmont Cragin', Decimal('10.8'), Decimal('18.7'), Decimal('14.6'), Decimal('37.3'), Decimal('37.3'), 15461, 70),
 (20, 'Hermosa', Decimal('6.9'), Decimal('20.5'), Decimal('13.1'), Decimal('41.6'), Decimal('36.4'), 15089, 71)]

## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [5]:
%sql select count(CASE_NUMBER) FROM "CHICAGO_CRIME_DATA";

 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[(533,)]

### Problem 2

##### List community areas with per capita income less than 11000.


In [6]:
%sql select distinct(COMMUNITY_AREA_NAME) from "CENSUS_DATA" where PER_CAPITA_INCOME < 11000;

 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[('Fuller Park',),
 ('Riverdale',),
 ('South Lawndale',),
 ('West Garfield Park',)]

### Problem 3

##### List all case numbers for crimes  involving minors?(children are not considered minors for the purposes of crime analysis)


In [7]:
%sql select CASE_NUMBER, DESCRIPTION  from "CHICAGO_CRIME_DATA" where LOWER(DESCRIPTION) like '%minor%';

 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[('HL266884', 'SELL/GIVE/DEL LIQUOR TO MINOR'),
 ('HK238408', 'ILLEGAL CONSUMPTION BY MINOR')]

### Problem 4

##### List all kidnapping crimes involving a child?


In [8]:
%%sql
SELECT DISTINCT(CASE_NUMBER), DESCRIPTION, PRIMARY_TYPE
FROM   "CHICAGO_CRIME_DATA"
WHERE  LOWER(DESCRIPTION) LIKE '%child%' AND LOWER(PRIMARY_TYPE) LIKE '%kidnapping%';

 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[('HN144152', 'CHILD ABDUCTION/STRANGER', 'KIDNAPPING')]

### Problem 5

##### What kinds of crimes were recorded at schools?


In [9]:
%%sql
SELECT CASE_NUMBER, PRIMARY_TYPE, LOCATION_DESCRIPTION 
FROM   "CHICAGO_CRIME_DATA"
WHERE  LOWER(LOCATION_DESCRIPTION ) LIKE '%school%';

 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[('HL353697', 'BATTERY', 'SCHOOL, PUBLIC, GROUNDS'),
 ('HL725506', 'BATTERY', 'SCHOOL, PUBLIC, BUILDING'),
 ('HP716225', 'BATTERY', 'SCHOOL, PUBLIC, BUILDING'),
 ('HH639427', 'BATTERY', 'SCHOOL, PUBLIC, BUILDING'),
 ('JA460432', 'BATTERY', 'SCHOOL, PUBLIC, GROUNDS'),
 ('HS200939', 'CRIMINAL DAMAGE', 'SCHOOL, PUBLIC, GROUNDS'),
 ('HK577020', 'NARCOTICS', 'SCHOOL, PUBLIC, GROUNDS'),
 ('HS305355', 'NARCOTICS', 'SCHOOL, PUBLIC, BUILDING'),
 ('HT315369', 'ASSAULT', 'SCHOOL, PUBLIC, GROUNDS'),
 ('HR585012', 'CRIMINAL TRESPA', 'SCHOOL, PUBLIC, GROUNDS'),
 ('HH292682', 'PUBLIC PEACE VI', 'SCHOOL, PRIVATE, BUILDING'),
 ('G635735', 'PUBLIC PEACE VI', 'SCHOOL, PUBLIC, BUILDING')]

### Problem 6

##### List the average safety score for each type of school.


In [10]:
%%sql
select "Elementary, Middle, or High School", AVG(DECIMAL(safety_score)) as average_safety
from "CHICAGO_PUBLIC_SCHOOLS" 
GROUP BY "Elementary, Middle, or High School";

 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[('ES', Decimal('49.52038369304556354916067146')),
 ('HS', Decimal('49.62352941176470588235294117')),
 ('MS', Decimal('48.00000000000000000000000000'))]

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [11]:
%%sql 
SELECT COMMUNITY_AREA_NAME, PERCENT_HOUSEHOLDS_BELOW_POVERTY FROM CENSUS_DATA
ORDER BY PERCENT_HOUSEHOLDS_BELOW_POVERTY DESC LIMIT 5;

 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[('Riverdale', Decimal('56.5')),
 ('Fuller Park', Decimal('51.2')),
 ('Englewood', Decimal('46.6')),
 ('North Lawndale', Decimal('43.1')),
 ('East Garfield Park', Decimal('42.4'))]

### Problem 8

##### Which community area is most crime prone?


In [28]:
%%sql
select COMMUNITY_AREA_NUMBER, count(*) as NUM_OF_CRIMES FROM CHICAGO_CRIME_DATA GROUP BY COMMUNITY_AREA_NUMBER ORDER BY NUM_OF_CRIMES DESC;

 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[(25, 43),
 (None, 43),
 (23, 22),
 (68, 21),
 (28, 16),
 (29, 16),
 (8, 15),
 (71, 14),
 (24, 13),
 (66, 12),
 (67, 12),
 (6, 11),
 (49, 11),
 (69, 11),
 (26, 10),
 (58, 10),
 (61, 10),
 (22, 9),
 (31, 9),
 (43, 9),
 (46, 9),
 (16, 8),
 (19, 8),
 (27, 8),
 (30, 8),
 (38, 8),
 (42, 8),
 (44, 8),
 (70, 8),
 (2, 7),
 (32, 7),
 (63, 7),
 (1, 6),
 (21, 6),
 (53, 6),
 (7, 5),
 (14, 5),
 (15, 5),
 (35, 5),
 (48, 5),
 (3, 4),
 (5, 4),
 (39, 4),
 (45, 4),
 (51, 4),
 (62, 4),
 (72, 4),
 (76, 4),
 (4, 3),
 (10, 3),
 (17, 3),
 (20, 3),
 (50, 3),
 (65, 3),
 (73, 3),
 (11, 2),
 (18, 2),
 (37, 2),
 (41, 2),
 (52, 2),
 (54, 2),
 (56, 2),
 (59, 2),
 (64, 2),
 (77, 2),
 (12, 1),
 (33, 1),
 (47, 1),
 (55, 1),
 (60, 1),
 (75, 1)]

In [24]:
%%sql 

SELECT  COMMUNITY_AREA_NAME FROM CENSUS_DATA 
WHERE   COMMUNITY_AREA_NUMBER = 25;



 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[('Austin',)]

Double-click **here** for a hint

<!--
Query for the 'community area number' that is most crime prone. Which means in the CHICAGO_CRIME_DATA find frequency for each community area number. i.e., select COMMUNITY_AREA_NUMBER, count(*) FROM CHICAGO_CRIME_DATA GROUP BY COMMUNITY_AREA_NUMBER ORDER BY DESC;
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [26]:
%%sql 
select COMMUNITY_AREA_NAME from CENSUS_DATA
WHERE hardship_index =(select MAX(hardship_index) from CENSUS_DATA);



 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[('Riverdale',)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [48]:

%sql select count(COMMUNITY_AREA_NUMBER)as NUM_OF_CRIMES FROM CHICAGO_CRIME_DATA GROUP BY COMMUNITY_AREA_NUMBER ORDER BY NUM_OF_CRIMES DESC LIMIT 1;




 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[(43,)]

In [55]:
%sql select COMMUNITY_AREA_NUMBER FROM CHICAGO_CRIME_DATA GROUP BY COMMUNITY_AREA_NUMBER ORDER BY count(COMMUNITY_AREA_NUMBER) DESC LIMIT 1;

 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[(25,)]

In [60]:
%%sql
select COMMUNITY_AREA_NUMBER, COMMUNITY_AREA_NAME FROM CENSUS_DATA
WHERE COMMUNITY_AREA_NUMBER in 
(select COMMUNITY_AREA_NUMBER FROM CHICAGO_CRIME_DATA GROUP BY COMMUNITY_AREA_NUMBER ORDER BY count(COMMUNITY_AREA_NUMBER) DESC LIMIT 1)

 * ibm_db_sa://gxn80469:***@54a2f15b-5c0f-46df-8954-7e38e612c2bd.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32733/BLUDB
Done.


[(25, 'Austin')]

Copyright © 2020 [cognitiveclass.ai](cognitiveclass.ai?utm_source=bducopyrightlink&utm_medium=dswb&utm_campaign=bdu). This notebook and its source code are released under the terms of the [MIT License](https://bigdatauniversity.com/mit-license?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ).


## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>

## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
| 2021-11-17 | 2.6     | Lakshmi           | Updated library                                |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>
